### PR TITLE
Disable the start review button on pull request when login user is pull request creator

### LINKED ISSUE #7289: Creator of pull request can chose to start review when commenting but never submit them
- Gitea version (or commit ref): b9d1fb6
- Git version: n/a
- Operating system: n/a
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes: https://try.gitea.io/themightychris/test/pulls/1/files#issuecomment-12063
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

If you create a pull request and then go to comment within that pull request, you're offered the option to just add a comment or start a review. If you start a review, every comment you post after that becomes part of the review and is held in a "Pending" status. But then you can't submit the review because you created the pull request and your comments can never be posted.

At the very least, if submitting a review is not an option then "start review" should not be an option when commenting. Ideally though, any admin or whitelisted review should be able to submit a review even if they created the pull request. I know this behavior was copied from GitHub per #4728 but its problematic: being the creator of a pull request does not mean you're the author of all/any of the commits within it. At the least it should be a branch option whether whitelisted reviewers can review their own pull requests. My use case is that I open develop->master pull requests for each release as the maintainer of the project and want to then review the release. It seems entirely common that maintainers on small teams might open the pull requests for others' pushed branches to create a review context, it seems silly that it should need to be a rule that maintainers wait for people to open their own PR in order to be able to start reviewing a branch

## Screenshots

image

### PR DESCRIPTION
**Pull request by lunny**
Will fix #7289

### CODE REVIEW

**Comment by silverwind** [comment]
I'd remove the button completely and ideally also reject the request server-side. Otherwise, a user could just remove the `disabled` attribute on the button to work around.

**Comment by lunny** [comment]
@silverwind As this situation, I don't think remove the buttons is better than disable them. We should let people know we have this feature, but because some reason it disabled.
And serverside check are already there. see https://github.com/go-gitea/gitea/blob/master/routers/repo/pull_review.go#L118

**Comment by silverwind** [comment]
Hmm maybe add a tooltip or `title` attribute to the button with a text on why it's disabled.

**Review by guillep2k — APPROVED** [review]
LGTM. Maybe a tooltip or message is useful, but I think it's reasonable to just disable the buttons. Starting a review for my own PR is at least confusing.

**Comment by guillep2k** [comment]
If a message is desired, I think a good place to put it is here:

image

Otherwise the floated div might be a hassle to deal with.

The message can be: "_You are not allowed to review your own PR._"
Also: "_You are not allowed to review your own PR, but you can comment on it._"

**Comment by lafriks** [comment]
Why should you not be able to comment on your own PR? You can not approve or reject but you should be able to start review and comment to it

**Comment by guillep2k** [comment]
> 
> 
> Why should you not be able to comment on your own PR? You can not approve or reject but you should be able to start review and comment to it

Of course you should be able to comment. Perhaps I didn't express myself correctly. I meant that commenting is the only thing that _without a doubt_ you should be able to do with your own PR. On the other hand, being able to approve or request changes (i.e. _review_), it's debatable.

**Comment by guillep2k** [comment]
> 
> 
> Why should you not be able to comment on your own PR? You can not approve or reject but you should be able to start review and comment to it

@lafriks To be clear about what's being worked on here:

If you're a reviewer, but not the original PR poster:
* **Comment**: enabled
* **Approve**: enabled
* **Request changes**: enabled

If you are the poster of the PR:
* **Comment**: enabled
* **Approve**: _disabled_
* **Request changes**: _disabled_

Is that OK with you?

**Comment by lafriks** [comment]
@guillep2k that was how it was initially actually, don't know at what point it has changed :)

File: templates/repo/diff/comment_form.tmpl [hunk 1]
```diff
@@ -30,16 +30,16 @@
 				{{else}}
 					{{if $.root.CurrentReview}}
 						<button name="is_review" value="true" type="submit"
```
Comment (6543) [inline templates/repo/diff/comment_form.tmpl]:
```suggestion
						{{$reviewCommentEnabled := (if and $.root.IsSigned ($.root.Issue.IsPoster $.root.SignedUser.ID)) }}
						<button name="is_review" value="true" type="submit"
```

File: templates/repo/diff/comment_form.tmpl [hunk 2]
```diff
@@ -30,16 +30,16 @@
...
-								class="ui submit green tiny button btn-add-comment">{{$.root.i18n.Tr "repo.diff.comment.add_review_comment"}}</button>
+								class="ui submit green tiny button btn-add-comment" {{if and $.root.IsSigned ($.root.Issue.IsPoster $.root.SignedUser.ID) }}disabled{{ end }}>{{$.root.i18n.Tr "repo.diff.comment.add_review_comment"}}</button>
```
Comment (6543) [inline templates/repo/diff/comment_form.tmpl]:
```suggestion
								class="ui submit green tiny button btn-add-comment {{if $reviewCommentEnabled}}poping up{{end}}" {{if $reviewCommentEnabled}}disabled data-content="{{$.i18n.Tr "repo.diff.comment.review_comment_disabled"{{ end }}>{{$.root.i18n.Tr "repo.diff.comment.add_review_comment"}}</button>
```

**Comment by 6543** [comment]
what's the current state?

and a idear: disabe buton and use "poping up" class with data-content="reason" so if mous is over botton a test pop up witch explain why it is disabled?

**Comment by lunny** [comment]
This PR need more work.

**Comment by 6543** [comment]
please resolve convlicts

**Comment by 6543** [comment]
whats the status?

this pull is wip? and #7289 is closed! ?

**Comment by 6543** [comment]
@lunny whatvif you as a pr creator like to create a comment reviw with multible code comments, youbstill need this fuction

**Comment by lunny** [comment]
Close as #7289 closed.

**Pull request closed without merging**